subt/artf_node.py

```diff
@@ -27,6 +28,26 @@
 }
 
 
+def check_results(result, result_cv):
+    ret = []
+    if len(result) == 0 or len(result_cv) == 0:
+        return ret
+    for r_cv in result_cv:
+        name_cv, score_cv, bbox = r_cv
+        bbox_x1, bbox_y1, bbox_x2, bbox_y2 = bbox
+        for r in result:
+            name, points = r
+            if name != name_cv:
+                continue
+            x = np.array([p[0] for p in points])
+            y = np.array([p[1] for p in points])
+            x_in_bbox = (x > bbox_x1) & (x < bbox_x2)  # arr of boolean values
+            y_in_bbox = (y > bbox_y1) & (y < bbox_y2)  # arr of boolean values
+            if sum(x_in_bbox & y_in_bbox) >= 1:  #at least one point is in the bbox
+                ret.append(r)
```

Comment:
Can bounding boxes from the CV detector overlap? If yes, we can end up reporting the same result twice.

Reply:
Yes, this is a weakness. However, the [result2report](https://github.com/robotika/osgar/blob/feature/detector-subt/subt/artf_node.py#L51) function then prepare the report for one object only. Yes, it should be improved.

Reply:
We can already think one step ahead and assume that we fix result2report at some point. What should we do about overlapping bounding boxes? I would be happier not to leave a trap here only to walk into it later.

Reply:
That trap should already be fixed. I forgot mention it. [f4ab2bd](https://github.com/robotika/osgar/pull/637/commits/f4ab2bdc15b35c8dd940472ff75ed08d6e777b3b)